Permission denied running plugin since packer 1.5.0

#### Overview of the Issue

Since Packer 1.5.0, we could not run vsphere-clone and vsphere-iso plugins anymroe.


#### Reproduction Steps

Install vsphere plugins https://github.com/jetbrains-infra/packer-builder-vsphere

Run a build.json using these plugins

### Packer version

Packer 1.5.0

### Operating system and Environment details

Linux 64 bits

### Log Fragments and crash.log files

```bash
$ PACKER_LOG=1 packer.io -only=vsphere-clone build.json
2019/12/20 11:24:36 [INFO] Packer version: 1.5.0 [go1.13.5 linux amd64]
2019/12/20 11:24:36 [DEBUG] Discovered plugin: vsphere-clone = /usr/local/bin/packer-builder-vsphere-clone.linux
2019/12/20 11:24:36 [DEBUG] Discovered plugin: vsphere-iso = /usr/local/bin/packer-builder-vsphere-iso
2019/12/20 11:24:36 using external builders [vsphere-clone vsphere-iso]
2019/12/20 11:24:36 Attempting to open config file: /home/user/.packerconfig
2019/12/20 11:24:36 [WARN] Config file doesn't exist: /home/user/.packerconfig
2019/12/20 11:24:36 Setting cache directory: /home/user/centos7_http/packer_cache
2019/12/20 11:24:36 Plugin could not be found at  (exec: "": executable file not found in $PATH). Checking same directory as executable.
2019/12/20 11:24:36 Current exe path: /usr/local/bin/packer.io
2019/12/20 11:24:36 Creating plugin client for path: /usr/local/bin
2019/12/20 11:24:36 Starting plugin: /usr/local/bin []string{"/usr/local/bin"}
Failed to initialize build 'vsphere-clone': error initializing builder 'vsphere-clone': fork/exec /usr/local/bin: permission denied
2019/12/20 11:24:36 [INFO] (telemetry) Finalizing.
2019/12/20 11:24:38 [WARN] (telemetry) Error finalizing report. This is safe to ignore. Post https://checkpoint-api.hashicorp.com/v1/telemetry/packer: context deadline exceeded
2019/12/20 11:24:38 waiting for all plugin processes to complete...
make: *** [Makefile:22: vsphere_validate] Error 1
$ echo $PATH
/usr/local/bin:/usr/bin:/bin:/usr/games:/home/user/.local/bin:/home/user/.local/bin
```

--- a/config.go
+++ b/config.go
@@ -131,12 +131,12 @@ func (c *config) discoverExternalComponents(path string) error {
 	if err != nil {
 		return err
 	}
-	for plugin, path := range pluginPaths {
-		newPath := path // this needs to be stored in a new variable for the func below
-		c.Builders[plugin] = func() (packer.Builder, error) {
+	for pluginName, pluginPath := range pluginPaths {
+		newPath := pluginPath // this needs to be stored in a new variable for the func below
+		c.Builders[pluginName] = func() (packer.Builder, error) {
 			return c.pluginClient(newPath).Builder()
 		}
-		externallyUsed = append(externallyUsed, plugin)
+		externallyUsed = append(externallyUsed, pluginName)
 	}
 	if len(externallyUsed) > 0 {
 		sort.Strings(externallyUsed)
@@ -148,12 +148,12 @@ func (c *config) discoverExternalComponents(path string) error {
 	if err != nil {
 		return err
 	}
-	for plugin, path := range pluginPaths {
-		newPath := path // this needs to be stored in a new variable for the func below
-		c.PostProcessors[plugin] = func() (packer.PostProcessor, error) {
+	for pluginName, pluginPath := range pluginPaths {
+		newPath := pluginPath // this needs to be stored in a new variable for the func below
+		c.PostProcessors[pluginName] = func() (packer.PostProcessor, error) {
 			return c.pluginClient(newPath).PostProcessor()
 		}
-		externallyUsed = append(externallyUsed, plugin)
+		externallyUsed = append(externallyUsed, pluginName)
 	}
 	if len(externallyUsed) > 0 {
 		sort.Strings(externallyUsed)
@@ -165,12 +165,12 @@ func (c *config) discoverExternalComponents(path string) error {
 	if err != nil {
 		return err
 	}
-	for plugin, path := range pluginPaths {
-		newPath := path // this needs to be stored in a new variable for the func below
-		c.Provisioners[plugin] = func() (packer.Provisioner, error) {
+	for pluginName, pluginPath := range pluginPaths {
+		newPath := pluginPath // this needs to be stored in a new variable for the func below
+		c.Provisioners[pluginName] = func() (packer.Provisioner, error) {
 			return c.pluginClient(newPath).Provisioner()
 		}
-		externallyUsed = append(externallyUsed, plugin)
+		externallyUsed = append(externallyUsed, pluginName)
 	}
 	if len(externallyUsed) > 0 {
 		sort.Strings(externallyUsed)
@@ -209,9 +209,9 @@ func (c *config) discoverSingle(glob string) (map[string]string, error) {
 		}
 
 		// Look for foo-bar-baz. The plugin name is "baz"
-		plugin := file[len(prefix):]
-		log.Printf("[DEBUG] Discovered plugin: %s = %s", plugin, match)
-		res[plugin] = match
+		pluginName := file[len(prefix):]
+		log.Printf("[DEBUG] Discovered plugin: %s = %s", pluginName, match)
+		res[pluginName] = match
 	}
 
 	return res, nil

--- a/config.go
+++ b/config.go
@@ -132,10 +132,9 @@ func (c *config) discoverExternalComponents(path string) error {
 		return err
 	}
 	for plugin, path := range pluginPaths {
-		plugin := plugin
-		path := path
+		newPath := path // this needs to be stored in a new variable for the func below
 		c.Builders[plugin] = func() (packer.Builder, error) {
-			return c.pluginClient(path).Builder()
+			return c.pluginClient(newPath).Builder()
 		}
 		externallyUsed = append(externallyUsed, plugin)
 	}
@@ -150,10 +149,9 @@ func (c *config) discoverExternalComponents(path string) error {
 		return err
 	}
 	for plugin, path := range pluginPaths {
-		plugin := plugin
-		path := path
+		newPath := path // this needs to be stored in a new variable for the func below
 		c.PostProcessors[plugin] = func() (packer.PostProcessor, error) {
-			return c.pluginClient(path).PostProcessor()
+			return c.pluginClient(newPath).PostProcessor()
 		}
 		externallyUsed = append(externallyUsed, plugin)
 	}
@@ -168,10 +166,9 @@ func (c *config) discoverExternalComponents(path string) error {
 		return err
 	}
 	for plugin, path := range pluginPaths {
-		plugin := plugin
-		path := path
+		newPath := path // this needs to be stored in a new variable for the func below
 		c.Provisioners[plugin] = func() (packer.Provisioner, error) {
-			return c.pluginClient(path).Provisioner()
+			return c.pluginClient(newPath).Provisioner()
 		}
 		externallyUsed = append(externallyUsed, plugin)
 	}

--- a/config.go
+++ b/config.go
@@ -131,10 +131,11 @@ func (c *config) discoverExternalComponents(path string) error {
 	if err != nil {
 		return err
 	}
-	for plugin := range pluginPaths {
+	for plugin, path := range pluginPaths {
 		plugin := plugin
+		path := path
 		c.Builders[plugin] = func() (packer.Builder, error) {
-			return c.pluginClient(pluginPaths[plugin]).Builder()
+			return c.pluginClient(path).Builder()
 		}
 		externallyUsed = append(externallyUsed, plugin)
 	}
@@ -148,10 +149,11 @@ func (c *config) discoverExternalComponents(path string) error {
 	if err != nil {
 		return err
 	}
-	for plugin := range pluginPaths {
+	for plugin, path := range pluginPaths {
 		plugin := plugin
+		path := path
 		c.PostProcessors[plugin] = func() (packer.PostProcessor, error) {
-			return c.pluginClient(pluginPaths[plugin]).PostProcessor()
+			return c.pluginClient(path).PostProcessor()
 		}
 		externallyUsed = append(externallyUsed, plugin)
 	}
@@ -165,10 +167,11 @@ func (c *config) discoverExternalComponents(path string) error {
 	if err != nil {
 		return err
 	}
-	for plugin := range pluginPaths {
+	for plugin, path := range pluginPaths {
 		plugin := plugin
+		path := path
 		c.Provisioners[plugin] = func() (packer.Provisioner, error) {
-			return c.pluginClient(pluginPaths[plugin]).Provisioner()
+			return c.pluginClient(path).Provisioner()
 		}
 		externallyUsed = append(externallyUsed, plugin)
 	}

